sdk/src/test/scala/com/horizen/SidechainNodeViewHolderTest.scala

```diff
@@ -448,4 +455,225 @@ class SidechainNodeViewHolderTest extends JUnitSuite
+   *  - check that first ModifiersProcessingResult message contain 2 applied blocks(1st, 2nd)
+   *  - check that second ModifiersProcessingResult message contain 4 applied blocks(3rd, 4th, 5th, 6th)
+   */
+  @Test
+  def remoteModifiersTwoSequences(): Unit = {
+    val block1 = generateNextSidechainBlock(genesisBlock, sidechainTransactionsCompanion, params)
+    val block2 = generateNextSidechainBlock(block1, sidechainTransactionsCompanion, params)
+    val block3 = generateNextSidechainBlock(block2, sidechainTransactionsCompanion, params)
+    val block4 = generateNextSidechainBlock(block3, sidechainTransactionsCompanion, params)
+    val block5 = generateNextSidechainBlock(block4, sidechainTransactionsCompanion, params)
+    val block6 = generateNextSidechainBlock(block5, sidechainTransactionsCompanion, params)
+
+    val firstRequestBlocks = Seq(block1, block2, block6)
+    val secondRequestBlocks = Seq(block3, block4, block5)
+    val correctSequence = Array(block1, block2, block3, block4, block5, block6)
+    var blockIndex = 0
+
+    val countDownController: CountDownLatchController = new CountDownLatchController(1)
+
+    // History appending check
+    Mockito.when(history.append(ArgumentMatchers.any[SidechainBlock])).thenAnswer(answer => {
+      Success(history -> ProgressInfo[SidechainBlock](None, Seq(), Seq(), Seq()))
+    })
+
+    Mockito.when(history.applicableTry(ArgumentMatchers.any[SidechainBlock])).thenAnswer(answer => {
+      val block: SidechainBlock = answer.getArgument(0)
+
+      if (block.id == correctSequence(blockIndex).id) {
+        if (blockIndex == 1) {
+          countDownController.countDown()
+        }
+
+        blockIndex += 1
+        Success(Unit)
+      } else
+        Failure(new RecoverableModifierError("Parent block is not in history yet"))
+    })
+
+    val eventListener = TestProbe()
+    actorSystem.eventStream.subscribe(eventListener.ref, classOf[ModifiersProcessingResult[(Seq[SidechainBlock], Seq[SidechainBlock])]])
+
+    mockedNodeViewHolderRef ! ModifiersFromRemote(firstRequestBlocks)
+    countDownController.await(3000)
+    Thread.sleep(1000)
+    mockedNodeViewHolderRef ! ModifiersFromRemote(secondRequestBlocks)
+
+    eventListener.fishForMessage(timeout.duration) {
+      case m =>
+        m match {
+          case ModifiersProcessingResult(applied, _) => {
+            assertTrue("Applyed block sequence is differ", applied.toSet.equals(Set(block1, block2)))
+            true
+          }
+          case _ => false
+        }
+    }
+
+    eventListener.fishForMessage(timeout.duration) {
+      case m =>
+        m match {
+          case ModifiersProcessingResult(applied, _) => {
+            assertTrue("Applyed block sequence is differ", applied.toSet.equals(Set(block3, block4, block5, block6)))
+            true
+          }
+          case _ => false
+        }
+    }
+  }
+
+  @Test
+  def remoteModifiersCacheClean(): Unit = {
```

Comment:
let's add test description as was did for the tests above. It is very helpful!

Reply:
Done.